scripts/globals/survival_guide.lua

```diff
@@ -0,0 +1,219 @@
+require("scripts/globals/settings")
+require("scripts/globals/survival_guide_map")
+
+dsp = dsp or {}
+dsp.survivalGuide = dsp.survivalGuide or {}
+
+-- Determines if the survival guide teleport cost is like if you had a Rhapsody in White key item. Does not affect UI! (Default: 0)
+local SURVIVAL_GUIDE_TELEPORT_COST_GIL = 1000
+local SURVIVAL_GUIDE_TELEPORT_COST_TABS = 50
+
+-- This is used for the NationTeleport save/get
+local SURVIVALGUIDE = 9
+local cutsceneID = 8500
+local tempMenuLayoutVar = "SgMenuLayout"
+
+local optionMap =
+{
+    TELEPORT = 1,
+    UNKNOWN = 2,
+    SET_MENU_LAYOUT = 3,
+    ADD_FAVORITE = 4,
+    REMOVE_FAVORITE = 5,
+    REPLACE_FAVORITE = 6,
+    TELEPORT_MENU = 7
+}
+
+-------------------------------------------------
+-- local functions
+-------------------------------------------------
+
+-- 1 = Original zones
+-- 2 = Zilart Zones
+-- 4 = Chains of Promethia Zones
+-- 8 = Treasures of Aht Urgahn
+-- 16 = Wings of the Goddess
+-- 2048 = Seekers of Auldoin
+local function getEnabledExpansions()
+    -- Original areas and zilart are always available.
+    local returnValue = 3
+
+    if ENABLE_COP == 1 then returnValue = returnValue + 4 end
+
+    if ENABLE_TOAU == 1 then returnValue = returnValue + 8 end
+
+    if ENABLE_WOTG == 1 then returnValue = returnValue + 16 end
+
+    if ENABLE_SOA == 1 then returnValue = returnValue + 2048 end
+
+    return returnValue
+end
+
+local function checkForRegisteredSurvivalGuide(player, guide, masks)
+    local group = guide.group
+    local mask = nil
+    local variableSuffix = ''
+    local hasRegisteredGuide = false
+
+    if guide.groupIndex == 32 then
+        mask = masks[group + 4]
+        variableSuffix = 'a'
+
+        -- This turns negative in the database for int32 max
+        hasRegisteredGuide = mask == (guide.groupMask * -1)
+    else
+        mask = masks[group]
+        hasRegisteredGuide = (bit.band(mask, guide.groupMask) ~= 0)
+    end
+
+    if not hasRegisteredGuide then
+        player:messageSpecial(zones[guide.zoneId].text.COMMON_SENSE_SURVIVAL)
+
+        local updatedValue = mask + guide.groupMask
+
+        if variableSuffix == 'a' then
+            updatedValue = guide.groupMask
+            masks[group + 4] = updatedValue
+        else
+            masks[group] = updatedValue
+        end
+
+        player:setVar("SgTeleportMask" .. group .. variableSuffix, updatedValue)
+
+        return false
+    end
+
+    return true
+end
+
+local function teleportMenuUpdate(player, option, teleportType, optionMap, var)
+    local choice = bit.band(option, 0xFF)
+
+    if choice == optionMap.SET_MENU_LAYOUT then
+        if bit.rshift(option, 16) == 1 then
+            -- Select by content/expansion
+            player:setVar(var, 1)
+        else
+            -- Select by region
+            player:setVar(var, 0)
```

Comment:
player:setVar changed to player:setCharVar. Righ @ibm2431 ?

Reply:
It did, but I wouldn't worry about updating this just yet due to wanting to change these to use char_unlock blobs.